src/common/arrstr.cpp

```diff
@@ -759,47 +756,52 @@ struct wxStringFragment
 
 wxStringFragment GetFragment(wxString& text)
 {
-    static const wxRegEx reSpaceOrPunct(wxS("^([[:space:]]|[[:punct:]])+"));
-    // Limit the length to make sure the value will fit into a wxUint64
-    static const wxRegEx reDigit(wxS("^[[:digit:]]{1,19}"));
-    static const wxRegEx reLetterOrSymbol("^[^[:space:]|[:punct:]|[:digit:]]+");
-
     if ( text.empty() )
         return wxStringFragment();
 
-    wxStringFragment fragment;
-    size_t           length = 0;
+    // the maximum length of a sequence of digits that
+    // can fit into wxUint64 when converted to a number
+    static const size_t maxDigitSequenceLength = 19;
```

Comment:
```suggestion
    static const ptrdiff_t maxDigitSequenceLength = 19;
```
should fix the build failures (although I didn't test it).